### PR TITLE
Adds unit tests for DBHandler and MainActivity.

### PR DESCRIPTION
**Pull request by DreamyPhobic**
Following tests are covered in this:
1. SQLite Database communications test
2. Test for connection with ODK Services Database
3. QRCode Scanner Test

### CODE REVIEW

File: app/src/androidTest/java/org/odk/odknotifications/Activities/MainActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.not;
+import static android.app.Instrumentation.ActivityResult;
+import static org.hamcrest.core.AllOf.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class MainActivityTest {
+
+    @Rule
+    public GrantPermissionRule grantPermissionRule = GrantPermissionRule.grant("android.permission.CAMERA");
```
Comment (linl33) [inline app/src/androidTest/java/org/odk/odknotifications/Activities/MainActivityTest.java]: The string literal can be replaced by `Manifest.permission.CAMERA`

File: app/src/androidTest/java/org/odk/odknotifications/Activities/MainActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    @Test
+    public void databaseConnectionTest(){
+        onView(allOf(withId(R.id.name_tv),not(withText("Bat"))));
+    }
+
+    @Test
+    public void qrCodeIntentTest(){
+        onView(withId(R.id.fab)).perform(click());
+        intended(allOf(
+                hasAction("com.google.zxing.client.android.SCAN")));
```
Comment (linl33) [inline app/src/androidTest/java/org/odk/odknotifications/Activities/MainActivityTest.java]: This literal can be replaced by [this constant](https://github.com/zxing/zxing/blob/ae6d7e55af0b62cd18b6ac775ce3b66caf889892/android/src/com/google/zxing/client/android/Intents.java#L39)

File: app/src/androidTest/java/org/odk/odknotifications/DatabaseCommunicator/DBHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+
+@RunWith(AndroidJUnit4.class)
+public class DBHandlerTest{
+
+    private DBHandler dbHandler;
+
+    @Before
+    public void setUp(){
+        dbHandler = new DBHandler(InstrumentationRegistry.getTargetContext(),null,null,1);
```
Comment (linl33) [inline app/src/androidTest/java/org/odk/odknotifications/DatabaseCommunicator/DBHandlerTest.java]: `getTargetContext` is deprecated, check the javadoc for replacement.

File: app/src/androidTest/java/org/odk/odknotifications/Activities/MainActivityTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasAction;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.isInternal;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.not;
+import static android.app.Instrumentation.ActivityResult;
+import static org.hamcrest.core.AllOf.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class MainActivityTest {
```
Comment (linl33) [inline app/src/androidTest/java/org/odk/odknotifications/Activities/MainActivityTest.java:31]: This is a test that interacts with the UI, it is a good practice to annotate it with `@LargeTest`

File: app/src/androidTest/java/org/odk/odknotifications/Activities/MainActivityTest.java (hunk 1 above)
Comment (linl33) [inline app/src/androidTest/java/org/odk/odknotifications/Activities/MainActivityTest.java]: The `MainActivity` also needs `READ_EXTERNAL_STORAGE`

**Review by linl33 — CHANGES_REQUESTED** [review]
Overall this is a good start. There are just a few minor things. Keep adding more tests and soon we will have a robust test suite. 

In the `DBHandlerTest`, there should be a `@Before` or `@After` method that cleans the db to prevent potential interaction between the tests. In general, it is a good idea to make sure the environment in which tests start in is consistent and predictable.

Another thing that will be a problem, but don't worry about it now, is setting up ODK-X Services for the tests. Since Services needs read/write external storage permissions, they need to be granted before the tests start.

**Comment by DreamyPhobic** [comment]
@linl33 @jbeorse I've made the required changes.

**Review by linl33 — APPROVED** [review]

**Pull request merged**